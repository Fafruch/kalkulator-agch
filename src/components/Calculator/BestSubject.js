import React from 'react'
import PropTypes from 'prop-types'

import { PRIMARY_SUBJECTS } from '../../constants/SubjectTypes'

const BestSubject = ({ maxScoreWrapper, subjectType }) => {
  if (!maxScoreWrapper) return null
  return (
    <div>
      <p>
        Maksymalny wynik dla {subjectType === PRIMARY_SUBJECTS ? 'przedmiotu głównego' : 'języka'} to
        &nbsp;<b>{maxScoreWrapper.computedScore}</b> pkt. dla przedmiotu
        &nbsp;<b>{maxScoreWrapper.subject.name === '' ? 'bez nazwy' : maxScoreWrapper.subject.name}</b>
        &nbsp;obliczony zgodnie z formułą
        &nbsp;{maxScoreWrapper.formula === 'A/B' ? 'A lub B' : maxScoreWrapper.formula}.
      </p>
    </div>
  )
}
BestSubject.propTypes = {
  maxScoreWrapper: PropTypes.object,
  subjectType: PropTypes.string.isRequired
}

export default BestSubject
