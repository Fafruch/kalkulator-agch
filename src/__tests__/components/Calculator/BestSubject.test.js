import React from 'react'
import { shallow } from 'enzyme'

import BestSubject from '../../../components/Calculator/BestSubject'

const setup = (setupProps = {}) => {
  const defaultProps = {
    bestSubject: {
      name: '',
      maxScore: {
        value: 0,
        formula: ''
      }
    },
    subjectsType: 'primary'
  }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<BestSubject {...props} />)

  return {
    props,
    wrapper
  }
}

describe('BestSubject component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })

  test('should render "bez nazwy" when no subject name provided', () => {
    const { wrapper } = setup()

    expect(wrapper.find('b').at(1).text()).toEqual('bez nazwy')
  })

  test('should render "języka" when subjectsType is "lingual"', () => {
    const { wrapper } = setup({ subjectsType: 'lingual' })

    expect(wrapper.contains('języka')).toEqual(true)
  })

  test('should render "A lub B" when formula is "A/B"', () => {
    const { wrapper } = setup({ bestSubject: { maxScore: { formula: 'A/B' } } })

    expect(wrapper.contains('A lub B')).toEqual(true)
  })
})
