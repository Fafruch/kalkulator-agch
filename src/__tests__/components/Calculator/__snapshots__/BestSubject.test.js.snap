// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestSubject component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BestSubject
    bestSubject={
        Object {
            "maxScore": Object {
              "formula": "",
              "value": 0,
            },
            "name": "",
          }
    }
    subjectsType="primary"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Maksymalny wynik dla 
        przedmiotu głównego
         to  
        <b />
         pkt. dla przedmiotu  
        <b>
                bez nazwy
        </b>
         obliczony zgodnie z formułą  
        
        .
</p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Maksymalny wynik dla ",
          "przedmiotu głównego",
          " to  ",
          <b />,
          " pkt. dla przedmiotu  ",
          <b>
            bez nazwy
</b>,
          " obliczony zgodnie z formułą  ",
          "",
          ".",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Maksymalny wynik dla ",
        "przedmiotu głównego",
        " to  ",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "b",
        },
        " pkt. dla przedmiotu  ",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "bez nazwy",
          },
          "ref": null,
          "rendered": "bez nazwy",
          "type": "b",
        },
        " obliczony zgodnie z formułą  ",
        "",
        ".",
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Maksymalny wynik dla 
          przedmiotu głównego
           to  
          <b />
           pkt. dla przedmiotu  
          <b>
                    bez nazwy
          </b>
           obliczony zgodnie z formułą  
          
          .
</p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Maksymalny wynik dla ",
            "przedmiotu głównego",
            " to  ",
            <b />,
            " pkt. dla przedmiotu  ",
            <b>
              bez nazwy
</b>,
            " obliczony zgodnie z formułą  ",
            "",
            ".",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Maksymalny wynik dla ",
          "przedmiotu głównego",
          " to  ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "b",
          },
          " pkt. dla przedmiotu  ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "bez nazwy",
            },
            "ref": null,
            "rendered": "bez nazwy",
            "type": "b",
          },
          " obliczony zgodnie z formułą  ",
          "",
          ".",
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
